torchaudio/datasets/speechcommands.py

```diff
@@ -79,9 +90,30 @@ def __init__(self,
                     download_url(url, root, hash_value=checksum, hash_type="md5")
                 extract_archive(archive, self._path)
 
-        walker = walk_files(self._path, suffix=".wav", prefix=True)
-        walker = filter(lambda w: HASH_DIVIDER in w and EXCEPT_FOLDER not in w, walker)
-        self._walker = list(walker)
+        def _load_list(*filenames):
+            output = []
+            for filename in filenames:
+                filepath = os.path.join(self._path, filename)
+                with open(filepath) as fileobj:
+                    output += [os.path.normpath(os.path.join(self._path, line.strip())) for line in fileobj]
+            return output
+
+        if subset == "validation":
+            self._walker = _load_list("validation_list.txt")
+        elif subset == "testing":
+            self._walker = _load_list("testing_list.txt")
+        elif subset == "training":
+            excludes = set(_load_list("validation_list.txt", "testing_list.txt"))
+            walker = walk_files(self._path, suffix=".wav", prefix=True)
+            self._walker = [
+                w for w in walker
+                if HASH_DIVIDER in w
+                and EXCEPT_FOLDER not in w
```

Comment:
`if HASH_DIVIDER in w and EXCEPT_FOLDER not in w` is redundant because files in background noises do not have `_nohash_` in filenames.

Reply:
I'll avoid changing the logic here to avoid breaking any codes, say if a user moved a file in background_noise folder/etc. I can look into it for a later PR.